src/models/history/tree.ts

```diff
@@ -58,7 +58,9 @@ export const Tree = types.model("Tree", {
           if (tiles.find(existingTile => existingTile.id === tileId)) {
             // This tile has already been added to our list
           } else {
-            const tile = resolveIdentifier(TileModel, document, tileId);
+            // HACK: the snapshot processor returns a type which isn't compatible with
+            // resolveIdentifier
+            const tile = resolveIdentifier(TileModel as IAnyModelType, document, tileId);
```

Comment:
Given that the main advantage of the `snapshotProcessor` is supposed to be better typing, but the net effect seems to be that it gets the types wrong so that we have to cast in places where we didn't used to have to cast, is it really an improvement? 🤷 And that's without getting into the bug that you found in which the underlying type is inexplicably modified.

Reply:
This is the only place the types are messed up that I remember. Using the other approach caused many more typing problems. For example `TileModel.create({title: "me"})` did not work. 

That bug is annoying but in this case we aren't using the underlying internal type, so it won't affect us.